server/services/offer.service.js

```diff
@@ -1,15 +1,36 @@
 import mongoose from 'mongoose';
+import format from 'date-fns/format';
 import Offer from '../models/offer.model';
 import { ErrorHandler } from '../helpers/errorHandler';
 import httpStatus from '../helpers/httpStatus';
 import { OFFER_STATUS } from '../helpers/constants';
 import { getUserById } from './user.service';
 import { getEnquiryById, approveEnquiry } from './enquiry.service';
+import { getPropertyById } from './property.service';
 
 const { ObjectId } = mongoose.Types.ObjectId;
 
 export const getOfferById = async (id) => Offer.findById(id).select();
 
+export const generateReferenceCode = async (propertyId) => {
+  const property = await getPropertyById(propertyId);
+  const initials = property.name
+    .match(/\b(\w)/g)
+    .join('')
+    .toUpperCase();
+  const numberSold = await Offer.countDocuments({ propertyId })
+    .then((count) => {
```

Comment:
This can be converted into 1 line.

```suggestion
    .then((count) => count + 1;
```